services/src/main/java/org/keycloak/services/resources/account/AccountRestService.java

```diff
@@ -275,6 +222,39 @@ public ResourcesService resources() {
 
     // TODO Federated identities
 
+    /**
+     * Returns the applications with the given id in the specified realm.
+     *
+     * @param clientId client id to search for
+     * @return application with the provided id
+     */
+    @Path("/applications/{clientId}")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
```

Comment:
@bs-matil I forgot to mention this change in my last review. Why did you add this endpoint for?

Reply:
looks like accidentally. Shall I open a PR for removal?

Reply:
Yes, please. Sorry about that. I added the comment but it got lost during the review ...

Reply:
Sorry too, I tought I will at least have one more time to squash and cleanup. Wasn't expecting a so fast merge after the history of this PR

Reply:
https://github.com/keycloak/keycloak/pull/7467
PR for cleanup